lib/src/r_tree/leaf_node.dart

```diff
@@ -19,27 +19,27 @@ part of r_tree;
 /// A [Node] that is a leaf node of the tree.  These are created automatically
 /// by [RTree] when inserting/removing items from the tree.
 class LeafNode<E> extends Node<E> {
-  List<RTreeDatum<E>> _items = [];
-  List<RTreeDatum<E>> get children => _items;
+  List<RTreeDatum<E>?> _items = [];
+  List<RTreeDatum<E>?> get children => _items;
```

Comment:
I feel like we want _items and children to be non nullable here? That way the search function below could return a non nullable Iterable .. it may be empty, but at least you wouldn't have nulls as elements in your returned list?